pkg/api/default.go

```diff
@@ -28,6 +28,214 @@ var DefaultVMSizeKubeArguments = map[VMSize]map[AgentPoolProfileRole]map[Reserve
 			SystemReserved: "cpu=500m,memory=512Mi",
 		},
 	},
+	StandardD8sV3: {
```

Comment:
I think we need to rework this.

each instance size has different mem, disk, cpu requirements.  We should probably make this easier by defaulting to a percentage of cpu and memory for instance types and also allow someone the ability to specify it.  That would simplify this a bit.

Thoughts?

Reply:
I like the idea of defaulting to a percentage of the available resources. +1